app/code/Magento/Store/Controller/Store/Redirect.php

```diff
@@ -53,12 +60,14 @@ public function __construct(
         StoreResolverInterface $storeResolver,
         \Magento\Framework\Session\Generic $session,
         \Magento\Framework\Session\SidResolverInterface $sidResolver,
-        HashGenerator $hashGenerator
+        HashGenerator $hashGenerator,
+        StoreManagerInterface $storeManager
```

Comment:
Please change it to `StoreManagerInterface $storeManager = null` for backward compatible
https://devdocs.magento.com/guides/v2.3/contributor-guide/backward-compatible-development/